### PR TITLE
Fix the communication issue between ks-devops and Jenkins

### PR DESCRIPTION
**Pull request by LinuxSuRen**
There're three changes you can find in this PR:

* provide a constant password for the admin user
* make the default security realm be local instead of LDAP
  * we will support OIDC later
* provide a correct server address for the KubeSphere token server

### CODE REVIEW

File: charts/ks-devops/charts/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -12,14 +12,37 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+        {{- end -}}
+    {{- end -}}
 {{- end -}}
+
+{{/*
+Returns the admin password
+https://github.com/helm/charts/issues/5167#issuecomment-619137759
+*/}}
+{{- define "jenkins.password" -}}
+  {{ if .Values.Master.AdminPassword -}}
```
Comment (JohnNiang) [inline charts/ks-devops/charts/jenkins/templates/_helpers.tpl:32]: May miss `-` after `{{`.

**Comment by LinuxSuRen** [comment]
hi @JohnNiang , please help me to review this PR. Thanks!

**Review by JohnNiang — COMMENTED** [review]
> hi @JohnNiang , please help me to review this PR. Thanks!

WIP.

File: charts/ks-devops/charts/jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (LinuxSuRen) [reply charts/ks-devops/charts/jenkins/templates/_helpers.tpl:32]: It should be ok. I copied it from [here](https://github.com/jenkinsci/helm-charts/blob/4c0a60c4812cdc5c1e8a63fba56d91dc7e5de2f7/charts/jenkins/templates/_helpers.tpl#L56).

File: charts/ks-devops/templates/NOTES.txt [hunk 2]
```diff
@@ -1,4 +1,11 @@
-1. Get the application URL by running these commands:
+{{if (and (eq .Values.jenkins.Master.ServiceType "NodePort") (not (empty .Values.jenkins.Master.NodePort)))}}
+echo "Visit http://{{.Values.jenkins.Master.HostName}}:{{.Values.jenkins.Master.NodePort}} to use your application"
+{{end}}
+
+1. Get your '{{ .Values.jenkins.Master.AdminUser }}' user password of Jenkins by running:
+  printf $(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Release.Name }}-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
```
Comment (JohnNiang) [inline charts/ks-devops/templates/NOTES.txt:6]: `printf` should not be here.

File: charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@ data:
       mode: EXCLUSIVE
       numExecutors: 5
       scmCheckoutRetryCount: 2
+      disableRememberMe: true
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml:11]: It's not. I set this due to the security reason.

File: charts/ks-devops/templates/NOTES.txt (hunk 2 above)
Comment (LinuxSuRen) [reply charts/ks-devops/templates/NOTES.txt:6]: Why not? It works well if you try to run the command `make install-chart`. You will get the output, and you an get the password with it.

Comment (JohnNiang) [reply charts/ks-devops/templates/NOTES.txt:6]: Last output and expected output:

```diff
1. Get your 'admin' user password of Jenkins by running:
-  printf $(kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
+ $(kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
```

File: charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JohnNiang) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml:11]: Got it.

File: charts/ks-devops/charts/jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (JohnNiang) [reply charts/ks-devops/charts/jenkins/templates/_helpers.tpl:32]: I know that's okay to work. But we shall keep the style as consistent as possible.

File: charts/ks-devops/templates/NOTES.txt (hunk 2 above)
Comment (LinuxSuRen) [reply charts/ks-devops/templates/NOTES.txt:6]: I didn't get your point.

image

It looks good to me from above sceenshot.

Comment (JohnNiang) [reply charts/ks-devops/templates/NOTES.txt:6]: I've updated the codeblock. IMO, the command `printf` is unnecessary from here.

```shell
➜  ~ kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode;echo
p101OeIYbNbeEFFVV5NYq9
```

Comment (LinuxSuRen) [reply charts/ks-devops/templates/NOTES.txt:6]: > Keyword `printf` should not be here.

`printf` can be here. But I will take your suggestion.

File: charts/ks-devops/charts/jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (LinuxSuRen) [reply charts/ks-devops/charts/jenkins/templates/_helpers.tpl:32]: Sure.

File: charts/ks-devops/templates/NOTES.txt (hunk 2 above)
Comment (LinuxSuRen) [reply charts/ks-devops/templates/NOTES.txt]: See the changes from https://github.com/LinuxSuRen/ks-devops/pull/34/commits/4e3cb7e24125007cdc9036627db3d052f7e3a80c

**Comment by JohnNiang** [comment]
LGTM

**Pull request merged**